### PR TITLE
Move docs to md

### PR DESCRIPTION
**Pull request by joaquimrocha**
- Generate docs compliant with our docs viewer
- docs: Create a "Gadgets" section and rename "Gadgets & Guides"

### CODE REVIEW

**Review by marga-kinvolk — APPROVED** [review]
LGTM

File: docs/gadgets/dns.md [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
-# Gadget dns
+---
+# Code generated by 'make generate-documentation'. DO NOT EDIT.
+title: Gadget dns
+---
 
 The dns gadget traces DNS requests.
 
-<h3 id="cr-example">Example CR</h3>
+### Example CR {#cr-example}
```
Comment (alban) [inline docs/gadgets/dns.md:8]: It seems the id is visible in the markdown. Could it be removed completely?


Screenshot from 2021-09-14 15-52-34

Comment (joaquimrocha) [reply docs/gadgets/dns.md:8]: This is advanced markdown. Apparently Github doesn't process it.
Feel free to remove it if you want.

Comment (joaquimrocha) [reply docs/gadgets/dns.md:8]: I have removed it.

**Review by alban — APPROVED** [review]

**Pull request merged**